pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -547,6 +546,9 @@ func doRequests(ctx context.Context, downstream Handler, reqs []Request, recordS
 			mtx.Lock()
 			resps = append(resps, requestResponse{req, resp})
 			mtx.Unlock()
+
+			queryStatistics.Merge(partialStats)
```

Comment:
We want to track stats even in case of error, so I would move this above, right after `resp, err := downstream.Do(childCtx, req)`